builtin/providers/aws/resource_aws_launch_configuration.go

```diff
@@ -364,23 +364,26 @@ func resourceAwsLaunchConfigurationCreate(d *schema.ResourceData, meta interface
 		createLaunchConfigurationOpts.BlockDeviceMappings = blockDevices
 	}
 
+	var id string
 	if v, ok := d.GetOk("name"); ok {
 		createLaunchConfigurationOpts.LaunchConfigurationName = aws.String(v.(string))
-		d.SetId(d.Get("name").(string))
+		id = v.(string)
 	} else {
 		hash := sha1.Sum([]byte(fmt.Sprintf("%#v", createLaunchConfigurationOpts)))
-		config_name := fmt.Sprintf("terraform-%s", base64.URLEncoding.EncodeToString(hash[:]))
-		log.Printf("[DEBUG] Computed Launch config name: %s", config_name)
-		createLaunchConfigurationOpts.LaunchConfigurationName = aws.String(config_name)
-		d.SetId(config_name)
+		configName := fmt.Sprintf("terraform-%s", base64.URLEncoding.EncodeToString(hash[:]))
+		log.Printf("[DEBUG] Computed Launch config name: %s", configName)
+		createLaunchConfigurationOpts.LaunchConfigurationName = aws.String(configName)
```

Comment:
`createLaunchConfigurationOpts.LaunchConfigurationName = aws.String(configName)` could be moved out of the `if/else` block and set with `id`, but just a nitpick